Crash when trying to open GPX with cgeo

### Describe your problem!

Current nightly crashes when I try to open a GPX from my file explorer (in order to import the caches).


### How to reproduce?

- Have a GPX or zipped GPX on your file system
- Use file explorer to tap the file and select to open with cgeo
-

### Actual result after these steps?

App crashes and starts up again on dashboard, no import action

### Expected result after these steps?

Show list selection to select list to import the GPX

### Reproducible

Yes

### c:geo Version

2021.10.23-NB1

### System information

```text
Samsung Galaxy Tab S4
Android 10
```


### Additional Information

```
0-23 11:15:03.844  9947  9947 E AndroidRuntime: FATAL EXCEPTION: main
10-23 11:15:03.844  9947  9947 E AndroidRuntime: Process: cgeo.geocaching, PID: 9947
10-23 11:15:03.844  9947  9947 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{cgeo.geocaching/cgeo.geocaching.CacheListActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.os.Bundle.getString(java.lang.String)' on a null object reference
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3623)
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3775)
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2261)
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:107)
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:237)
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8107)
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
10-23 11:15:03.844  9947  9947 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.os.Bundle.getString(java.lang.String)' on a null object reference
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	at cgeo.geocaching.CacheListActivity.onCreate(CacheListActivity.java:592)
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7957)
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7946)
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1307)
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3598)
10-23 11:15:03.844  9947  9947 E AndroidRuntime: 	... 11 more
10-23 11:15:03.844  9947  9947 W cgeo    : [main] UncaughtException
10-23 11:15:03.844  9947  9947 W cgeo    : java.lang.RuntimeException: Unable to start activity ComponentInfo{cgeo.geocaching/cgeo.geocaching.CacheListActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.os.Bundle.getString(java.lang.String)' on a null object reference
10-23 11:15:03.844  9947  9947 W cgeo    : 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3623)
10-23 11:15:03.844  9947  9947 W cgeo    : 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3775)
10-23 11:15:03.844  9947  9947 W cgeo    : 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
10-23 11:15:03.844  9947  9947 W cgeo    : 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
10-23 11:15:03.844  9947  9947 W cgeo    : 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
10-23 11:15:03.844  9947  9947 W cgeo    : 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2261)
10-23 11:15:03.844  9947  9947 W cgeo    : 	at android.os.Handler.dispatchMessage(Handler.java:107)
10-23 11:15:03.844  9947  9947 W cgeo    : 	at android.os.Looper.loop(Looper.java:237)
10-23 11:15:03.844  9947  9947 W cgeo    : 	at android.app.ActivityThread.main(ActivityThread.java:8107)
10-23 11:15:03.844  9947  9947 W cgeo    : 	at java.lang.reflect.Method.invoke(Native Method)
10-23 11:15:03.844  9947  9947 W cgeo    : 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
10-23 11:15:03.844  9947  9947 W cgeo    : 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
10-23 11:15:03.844  9947  9947 W cgeo    : Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.os.Bundle.getString(java.lang.String)' on a null object reference
10-23 11:15:03.844  9947  9947 W cgeo    : 	at cgeo.geocaching.CacheListActivity.onCreate(CacheListActivity.java:592)
10-23 11:15:03.844  9947  9947 W cgeo    : 	at android.app.Activity.performCreate(Activity.java:7957)
10-23 11:15:03.844  9947  9947 W cgeo    : 	at android.app.Activity.performCreate(Activity.java:7946)
10-23 11:15:03.844  9947  9947 W cgeo    : 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1307)
10-23 11:15:03.844  9947  9947 W cgeo    : 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3598)
10-23 11:15:03.844  9947  9947 W cgeo    : 	... 11 more
```

--- a/main/src/cgeo/geocaching/CacheListActivity.java
+++ b/main/src/cgeo/geocaching/CacheListActivity.java
@@ -589,7 +589,9 @@ public class CacheListActivity extends AbstractListActivity implements FilteredA
         }
 
         if (isInvokedFromAttachment()) {
-            listNameMemento.rememberTerm(extras.getString(Intents.EXTRA_NAME));
+            if (extras != null && !StringUtils.isBlank(extras.getString(Intents.EXTRA_NAME))) {
+                listNameMemento.rememberTerm(extras.getString(Intents.EXTRA_NAME));
+            }
             importGpxAttachement();
         }
     }

